### PR TITLE
Feature/send invitation

### PR DESCRIPTION
**Pull request by mattdean-digicatapult**
# Pull Request

## Checklist
- [x] Have you read Digital Catapult's [Code of Conduct](https://github.com/digicatapult/.github/blob/main/CODE_OF_CONDUCT.md)?
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] New and existing unit tests pass locally with my changes.

## PR Type

Please delete options that are irrelevant.

- [ ] Bug Fix
- [x] Feature
- [ ] Documentation Update
- [ ] Code style update (formatting, local variables)
- [ ] Breaking Change (fix or feature that would cause existing functionality to change)

## Linked tickets

[VR-59](https://digicatapult.atlassian.net/browse/VR-59)

## High level description

Adds connection establishment invite emails

## Detailed description

This is quite a significant PR that adds a lot of new things including:
* new database structures to accomodate invite details
* refined logic for the new connection form process including various error paths
* generation of the out of band invite in the cloudagent
* generation of the second factor PIN number and stroage in the db as an argon2 hash
* unit tests for all the new connection controllers. These were previously missed
* integration tests for creating the new connection

## Describe alternatives you've considered

PR is implemented as described in the ticket.

## Operational impact

New environment variables will need to be configured but as there's no deployment yet this doesn't have any impact

## Additional context

N/A

[VR-59]: https://digicatapult.atlassian.net/browse/VR-59?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -1,5 +1,8 @@
 {
   "typescript.tsdk": "node_modules/typescript/lib",
   "typescript.enablePromptUseWorkspaceTsdk": true,
-  "svg.preview.background": "custom"
+  "svg.preview.background": "custom",
+  "cSpell.words": [
+    "Cloudagent"
+  ]
 }
```
Comment (n3op2) [inline .vscode/settings.json:8]: Does this needs to be included or something .gitignore?

File: package.json [hunk 2]
```diff
@@ -22,7 +22,7 @@
     "db:cmd": "node --import=tsimp/import ./node_modules/.bin/knex",
     "db:migrate": "npm run db:cmd -- migrate:latest --knexfile src/models/db/knexfile.ts",
     "db:rollback": "npm run db:cmd -- migrate:rollback --knexfile src/models/db/knexfile.ts",
-    "db:seed": "npm run db:cmd -- seed:run",
+    "db:seed": "npm run db:cmd -- seed:run --knexfile src/models/db/knexfile.ts",
```
Comment (n3op2) [inline package.json:26]: should have updated

File: src/models/__tests__/helpers/mockCloudagent.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  let agent: MockAgent
+  beforeEach(function () {
+    originalDispatcher = getGlobalDispatcher()
+    agent = new MockAgent()
+    setGlobalDispatcher(agent)
+
+    const client = agent.get(env.get('CLOUDAGENT_ADMIN_ORIGIN'))
+    client
+      .intercept({
+        path: `/oob/create-invitation`,
```
Comment (n3op2) [inline src/models/__tests__/helpers/mockCloudagent.ts:20]: yay! first interaction (even if it's a mock)

**Review by n3op2 — APPROVED** [review]

File: .vscode/settings.json (hunk 1 above)
Comment (mattdean-digicatapult) [reply .vscode/settings.json]: removed

File: package.json (hunk 2 above)
Comment (mattdean-digicatapult) [reply package.json:26]: easy to miss things

File: src/models/__tests__/helpers/mockCloudagent.ts (hunk 3 above)
Comment (mattdean-digicatapult) [reply src/models/__tests__/helpers/mockCloudagent.ts:20]: works for reals as well!

**Review by n3op2 — APPROVED** [review]

File: .vscode/settings.json [hunk 4]
```diff
@@ -1,5 +1,4 @@
 {
   "typescript.tsdk": "node_modules/typescript/lib",
-  "typescript.enablePromptUseWorkspaceTsdk": true,
-  "svg.preview.background": "custom"
+  "typescript.enablePromptUseWorkspaceTsdk": true
 }
```
Comment (n3op2) [inline .vscode/settings.json:4]: .vscode still included

Comment (mattdean-digicatapult) [reply .vscode/settings.json]: yep reverted

**Review by n3op2 — APPROVED** [review]

**Pull request merged**